File: app/src/main/java/com/example/myapplication/SecondActivity.java
package com.example.myapplication;

import android.content.Intent;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import org.w3c.dom.Text;

public class SecondActivity extends AppCompatActivity implements View.OnClickListener {
    private static final String PASS = "PASS";
    private String z = " ";
    String save;
    TextView textView;
    EditText editText;
    final static String LANDSAVE = "LANDSAVE";
    Button ButtonOne;
    Button ButtonTwo;
    Button ButtonThree;
    Button ButtonFive;
    Button ButtonSix;
    Button ButtonFour;
    Button ButtonClear;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        Button b = findViewById(R.id.backToTheFirstScreen);
        b.setOnClickListener(this);
        ButtonTwo = findViewById(R.id.button_2);
        ButtonOne = findViewById(R.id.button_1);
        ButtonThree = findViewById(R.id.button_3);
        ButtonFour = findViewById(R.id.button_4);
        ButtonFive = findViewById(R.id.button_5);
        ButtonSix = findViewById(R.id.button_6);
        ButtonClear = findViewById(R.id.buttonClear);
        editText = findViewById(R.id.editText);
        textView = findViewById(R.id.textView);
        if(savedInstanceState!=null){
            textView.setText(savedInstanceState.getString(LANDSAVE));
        }

        View.OnClickListener oclBtn1 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Edit(findViewById(R.id.button_1));
            }
        };
        ButtonOne.setOnClickListener(oclBtn1);
        if (editText.length()>2){

        }
        View.OnClickListener oclBtn2 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Edit(findViewById(R.id.button_2));
            }
        };
        ButtonTwo.setOnClickListener(oclBtn2);
        View.OnClickListener oclBtn3 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Edit(findViewById(R.id.button_3));
            }
        };
        ButtonThree.setOnClickListener(oclBtn3);
        View.OnClickListener oclBtn4 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Edit(findViewById(R.id.button_4));
            }
        };
        ButtonFour.setOnClickListener(oclBtn4);
        View.OnClickListener oclBtn5 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Edit(findViewById(R.id.button_5));
            }
        };
        ButtonFive.setOnClickListener(oclBtn5);
        View.OnClickListener oclBtn6 = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Edit(findViewById(R.id.button_6));
            }
        };
        ButtonSix.setOnClickListener(oclBtn6);
        View.OnClickListener oclBtnclear = new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                textView.setText(null);
            }
        };
        ButtonClear.setOnClickListener(oclBtnclear);


        editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                TextView textView = findViewById(R.id.textView);
                textView.setText(s);

            }

            @Override
            public void afterTextChanged(Editable s) {
                if (editText.length() > 2) {
                    StringBuilder builder = new StringBuilder(s);
                    builder.deleteCharAt(1);
                    builder.deleteCharAt(0);
                    editText.setText(builder);
                    editText.setSelection(1);
                }

            }
        });
    }
    @Override
    public void onSaveInstanceState(Bundle outState) {
        String save = textView.getText().toString();
        outState.putString("LANDSAVE", save);
    }
    public void Edit(Button f) {
        if (textView.length() <= 2) {
            textView.setText(textView.getText() + f.getText().toString());
        }
        if (textView.length() > 2) {
            StringBuilder build = new StringBuilder(textView.getText().toString());
            build.deleteCharAt(0);
            build.toString();
            textView.setText(build.substring(1));
        }
    }

public void onClick(View view) {
        Intent intent = new Intent(this, MainActivity.class);
        z=textView.getText().toString();
        intent.putExtra(PASS, z);
        setResult(RESULT_OK, intent);
        this.finish();
        }
}

File: app/src/main/java/com/example/myapplication/SecondActivity.kt
package com.example.myapplication

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.example.myapplication.databinding.ActivitySecondBinding


class SecondActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySecondBinding
    private var result = " "

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivitySecondBinding.inflate(layoutInflater)
        setContentView(binding.root)
        if (savedInstanceState != null) {
            binding.tvResultTitle.text = savedInstanceState.getString(LAND_SAVE)
        }

        initClickListeners()

        binding.etResult.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                if (binding.etResult.length() > 2) {
                    binding.etResult.setText(s.toString().drop(1))
                    binding.etResult.setSelection(1)
                }
                if (binding.etResult.text.isNotBlank()) checkAndSetResult(binding.etResult.text.last().toString())
            }

            override fun afterTextChanged(s: Editable) {}
        })
    }

    public override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString(LAND_SAVE, binding.etResult.text.toString())
    }

    private fun checkAndSetResult(text: String) {
        when {
            binding.tvResultTitle.length() == 2 -> binding.tvResultTitle.text =
                String.format("%s%s", binding.tvResultTitle.text.drop(1), text)
            else ->  binding.tvResultTitle.text = String.format("%s%s",  binding.tvResultTitle.text, text)
        }
    }

    private fun initClickListeners() {
        binding.buttonOne.setOnClickListener { checkAndSetResult(binding.buttonOne.text.toString()) }
        binding.buttonTwo.setOnClickListener { checkAndSetResult( binding.buttonTwo.text.toString()) }
        binding.buttonThree.setOnClickListener { checkAndSetResult(binding.buttonThree.text.toString()) }
        binding.buttonFour.setOnClickListener { checkAndSetResult(binding.buttonFour.text.toString()) }
        binding.buttonFive.setOnClickListener { checkAndSetResult( binding.buttonFive.text.toString()) }
        binding.buttonSix.setOnClickListener { checkAndSetResult( binding.buttonSix.text.toString()) }

        binding.btClear.setOnClickListener { binding.tvResultTitle.text = ""
        binding.etResult.text.clear()
        }

        binding.btBack.setOnClickListener {
            val intent = Intent(this, MainActivity::class.java)
            result = binding.tvResultTitle.text.toString()
            intent.putExtra(PASS, result)
            setResult(RESULT_OK, intent)
            finish()
        }
    }

}
